DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs

```diff
@@ -108,29 +49,31 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
         /// <returns></returns>
         public async Task<DeviceModel> AddDeviceAsync(DeviceModel device)
         {
-            string deviceId = device.DeviceProperties.DeviceID;
-            DeviceModel existingDevice = await this.GetDeviceAsync(deviceId);
+            if (string.IsNullOrEmpty(device.id))
+            {
+                device.id = Guid.NewGuid().ToString();
+            }
 
+            var deviceId = device.id;
```

Comment:
The id of the document in DocumentDB is NOT DeviceProperties.DeviceID. It's actually Device.id. Unfortunately that's the model being adopted today and to retain backcompat this will have to be maintained.